server/pruning.go

```diff
@@ -6,23 +6,22 @@ import (
 
 	"github.com/spf13/cast"
 
+	pruningtypes "github.com/cosmos/cosmos-sdk/pruning/types"
 	"github.com/cosmos/cosmos-sdk/server/types"
-	"github.com/cosmos/cosmos-sdk/store"
-	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 )
 
 // GetPruningOptionsFromFlags parses command flags and returns the correct
 // PruningOptions. If a pruning strategy is provided, that will be parsed and
 // returned, otherwise, it is assumed custom pruning options are provided.
-func GetPruningOptionsFromFlags(appOpts types.AppOptions) (storetypes.PruningOptions, error) {
+func GetPruningOptionsFromFlags(appOpts types.AppOptions) (pruningtypes.PruningOptions, error) {
 	strategy := strings.ToLower(cast.ToString(appOpts.Get(FlagPruning)))
 
 	switch strategy {
-	case storetypes.PruningOptionDefault, storetypes.PruningOptionNothing, storetypes.PruningOptionEverything:
-		return storetypes.NewPruningOptionsFromString(strategy), nil
+	case pruningtypes.PruningOptionDefault, pruningtypes.PruningOptionNothing, pruningtypes.PruningOptionEverything:
+		return pruningtypes.NewPruningOptionsFromString(strategy), nil
 
-	case storetypes.PruningOptionCustom:
-		opts := storetypes.NewPruningOptions(
+	case pruningtypes.PruningOptionCustom:
+		opts := pruningtypes.NewCustomPruningOptions(
```

Comment:
There's no need to put `Custom` in the constructor IMO

Reply:
What do you suggest changing it to?

Unfortunately, we cannot have functions with the same name and different parameters in Go. We already have `NewPruningOptions` that has different parameters so we need to choose a name other than that

Reply:
DIdn't realize this. Ignore my comment :p